source/georef/street_network.cpp

```diff
@@ -242,16 +241,77 @@ PathFinder::crow_fly_find_nearest_stop_points(navitia::time_duration radius,
                                               const proximitylist::ProximityList<type::idx_t>& pl) {
     // Searching for all the elements that are less than radius meters awyway in crow fly
     float crow_fly_dist = radius.total_seconds() * speed_factor * georef::default_speed[mode];
-
     return pl.find_within(start_coord, crow_fly_dist);
 }
 
+
+struct ProjectionGetterByCache{
+    const nt::Mode_e mode;
+    const std::vector<GeoRef::ProjectionByMode>& projection_cache;
+    const georef::ProjectionData& operator()(const routing::SpIdx& idx) const{
+        return projection_cache[idx.val][mode];
+    }
+};
+
+struct ProjectionGetterOnFly{
+    const GeoRef& geo_ref;
+    const nt::idx_t& offset;
+    const georef::ProjectionData operator()(const type::GeographicalCoord& coord) const{
+        return georef::ProjectionData{coord, geo_ref, offset, geo_ref.pl};
+    }
+};
+
+static routing::SpIdx get_id(routing::SpIdx idx) { return idx; }
+static std::string get_id(const type::GeographicalCoord& coord) { return coord.uri(); }
+
+template<typename K, typename U, typename G>
+boost::container::flat_map<K, navitia::time_duration>
+PathFinder::start_dijkstra_and_fill_duration_map(const navitia::time_duration& radius,
+        const std::vector<U>& destinations,
+        const G& projection_getter){
+    boost::container::flat_map<K, navitia::time_duration> result;
+    start_distance_dijkstra(radius);
+#ifdef _DEBUG_DIJKSTRA_QUANTUM_
+    dump_dijkstra_for_quantum(starting_edge);
+#endif
+    for (const auto& dest: destinations) {
+        const auto projection = projection_getter(dest);
```

Comment:
1. test projection before launching dijkstra 
2. stop the dijkastra when all destinations are arrived